src/interactive/azext_interactive/azclishell/recommendation.py

```diff
@@ -237,59 +243,86 @@ def get_recommend_from_api(command_list, type, top_num=5, error_info=None):  # p
     if 'data' in response.json():
         recommends = response.json()['data']
 
-    return recommends
-
-
-def send_feedback(option_idx, latest_commands, processed_exception=None, recommends=None, rec=None):
-    feedback_data = ['1', str(option_idx)]
-
-    trigger_commands = latest_commands[-1]
+    if 'dynamic_api_version' in response.json():
+        api_version = response.json()['dynamic_api_version']
+    return recommends, api_version
+
+
+def send_feedback(option_idx, latest_commands, processed_exception=None, recommends=None, accepted_recommend=None,
+                  api_version=None):
+    # initialize feedback data
+    feedback_data = {}
+    feedback_data["data_prefix"] = None
+    feedback_data["option_idx"] = None
+    feedback_data["trigger_commands"] = None
+    feedback_data["processed_exception"] = None
+    feedback_data["all_recommendations"] = None
+    feedback_data["recommendations_source_list"] = None
+    feedback_data["recommendations_type_list"] = None
+    feedback_data["accepted_recommend_source"] = None
+    feedback_data["accepted_recommend_type"] = None
+    feedback_data["accepted_recommend"] = None
+    feedback_data["is_personalized"] = None
```

Comment:
Since Python's dict is dynamic, do we need to initialize it?

Reply:
Received, already modified in https://github.com/wu5bocheng/azure-cli-extensions/pull/4/commits/9a6973723d19d70e75f87ae9325252100671ab95 and https://github.com/wu5bocheng/azure-cli-extensions/pull/4/commits/70cdd9c2f7d74fcb8a099a3b45c5cd3c02b083dd